sdk/java-sdk-spring/src/test/java/kalix/spring/ComponentClientTest.java

```diff
@@ -0,0 +1,269 @@
+    restKalixClient = new RestKalixClientImpl(messageCodec);
+    componentClient = new ComponentClient(restKalixClient);
+  }
+
+  @Test
+  public void shouldReturnDeferredCallForSimpleGETRequest() throws InvalidProtocolBufferException {
+    //given
+    var action = descriptorFor(GetWithoutParam.class, messageCodec);
+    restKalixClient.registerComponent(action.serviceDescriptor());
+    var targetMethod = action.serviceDescriptor().findMethodByName("Message");
+
+    //when
+    RestDeferredCall<Any, Message> call = (RestDeferredCall<Any, Message>) componentClient.forAction().call(GetWithoutParam::message);
+
+    //then
+    assertThat(call.fullServiceName()).isEqualTo(targetMethod.getService().getFullName());
+    assertThat(call.methodName()).isEqualTo(targetMethod.getName());
+    assertMethodParamsMatch(targetMethod, call.message());
+  }
+
+  @Test
+  public void shouldReturnDeferredCallForGETRequestWithParam() throws InvalidProtocolBufferException {
+    //given
+    var action = descriptorFor(GetWithOneParam.class, messageCodec);
+    restKalixClient.registerComponent(action.serviceDescriptor());
+    var targetMethod = action.serviceDescriptor().findMethodByName("Message");
+    String param = "a b&c@d";
+
+    //when
+    RestDeferredCall<Any, Message> call = (RestDeferredCall<Any, Message>) componentClient.forAction()
+        .call(GetWithOneParam::message)
+        .params(param);
+
+    //then
+    assertThat(call.fullServiceName()).isEqualTo(targetMethod.getService().getFullName());
+    assertThat(call.methodName()).isEqualTo(targetMethod.getName());
+    assertMethodParamsMatch(targetMethod, call.message(), param);
+  }
+
+  @Test
+  public void shouldReturnDeferredCallForGETRequestWithTwoParams() throws InvalidProtocolBufferException {
+    //given
+    var action = descriptorFor(GetClassLevel.class, messageCodec);
+    restKalixClient.registerComponent(action.serviceDescriptor());
+    var targetMethod = action.serviceDescriptor().findMethodByName("Message");
+    String param = "a b&c@d";
+    Long param2 = 2L;
+
+    //when
+    RestDeferredCall<Any, Message> call = (RestDeferredCall<Any, Message>) componentClient.forAction()
+        .call(GetClassLevel::message)
+        .params(param, param2);
+
+    //then
+    assertThat(call.fullServiceName()).isEqualTo(targetMethod.getService().getFullName());
+    assertThat(call.methodName()).isEqualTo(targetMethod.getName());
+    assertMethodParamsMatch(targetMethod, call.message(), param, param2);
+  }
+
+  @Test
+  public void shouldReturnDeferredCallForGETRequestWithTwoPathParamsAnd2ReqParams() throws InvalidProtocolBufferException {
+    //given
+    var action = descriptorFor(GetClassLevel.class, messageCodec);
+    restKalixClient.registerComponent(action.serviceDescriptor());
+    var targetMethod = action.serviceDescriptor().findMethodByName("Message2");
+    String param = "a b&c@d";
+    Long param2 = 2L;
+    String param3 = "!@!#$%^%++___";
```

Comment:
looks like you are mad 😄

Reply:
I spent a fair amount of time on double encoding issues:)